oar/core/jenkins_helper.py

```diff
@@ -137,3 +101,100 @@ def get_job_status(self, url, job_name, build_number):
+            return False
+        except Exception as e:
+            raise JenkinsHelperException(
+                f"check build status failed"
+            ) from e
+
+    def call_build_job(self,job_name):
+        """
+        trigger build job and return build url
+        """  
+        try: 
+            if(job_name == "image-consistency-check"):
+                parameters_value = {
+                    "VERSION": self.vversion,
+                    "ERRATA_NUMBERS": self.errata_numbers,
+                    "PAYLOAD_URL": self.pull_spec,
+                }
+            elif(job_name == "Stage-Pipeline"):
+                parameters_value = {
+                    "VERSION": self.version,
+                    "METADATA_AD": self.metadata_ad,
+                    "PULL_SPEC": self.pull_spec,
+                }
+                job_name = "zstreams/" +job_name
+            else:
+                logger.info("please provide correct job name")
+            res = self.server.build_job(
+                job_name,
+                parameters=parameters_value,
+            )
+            queue_item_info = self.server.get_queue_item(res)
+            queue_item_link = self._cs.get_jenkins_server() +"/queue/item/"+ str(res) +"/api/json?depth=0"
+            if(queue_item_info['blocked']):
+                return f"a new job triggered by you is pending, you can check queue item link: {queue_item_link}"
+            else: 
+                if(self.poll_url(queue_item_link)):
+                    return queue_item_info["executable"]["url"]
+                else:
+                    raise JenkinsHelperException(f"visit {queue_item_link} failed")
+        except jenkins.JenkinsException as ej:
+            raise JenkinsHelperException(
+                f"call {job_name} job failed"
+            ) from ej
+    
+    def init_jenkins_server(self):
+        try: 
+            return jenkins.Jenkins(
+                self._cs.get_jenkins_server(),
+                username=self._cs.get_jenkins_username(),
+                password=self._cs.get_jenkins_token(),
+            ) 
+        except jenkins.JenkinsException as ej:
+            raise JenkinsHelperException(
+                "failed to init jenkins server"
+            ) from ej
+
+    def visit_url(self, url):
+        """
+        visit url
+        """
+        try:
+            response = requests.get(url)
+            return True if response.status_code == 200 else False
+        except requests.exceptions.RequestException as e:
+            if isinstance(e, requests.exceptions.Timeout):
+                print("requests timeout")
+            else:
+                logger.warn(f"visit {url}, error info{e}")
+                return False
+
+    def poll_url(self, url):
+        """
+        visit url at regular intervals
+
+        """
+        res = self.visit_url(url)
+        if res:
+            return True
+        else:
+            global counter
+            if(counter < loop_counts):
```

Comment:
@wewang58 exec polling logic in one thread does not make sense, we just need to impl a simple poller with for loop.